src/vm/wks/CMakeLists.txt

```diff
@@ -1,67 +1,116 @@
+
 if (WIN32)
   add_precompiled_header(common.h ../common.cpp VM_SOURCES_WKS)
   # mscorlib.cpp does not compile with precompiled header file
   set_source_files_properties(../mscorlib.cpp PROPERTIES COMPILE_FLAGS "/Y-") 
   # .c file cannot use cpp precompiled header 
   set_source_files_properties(../microsoft.comservices_i.c PROPERTIES COMPILE_FLAGS "/Y-") 
 
-  # asm files require preprocessing using cl.exe on arm64
+  # asm files require preprocessing using cl.exe on arm32 and arm64
   if(CLR_CMAKE_PLATFORM_ARCH_ARM64)
+
       foreach(ASM_FILE ${VM_SOURCES_WKS_ARCH_ASM})
+
+          # Preprocess each asm source file
           get_filename_component(name ${ASM_FILE} NAME_WE)
           set(ASM_PREPROCESSED_FILE ${CMAKE_CURRENT_BINARY_DIR}/${name}.asm)
           preprocess_def_file(${ASM_FILE} ${ASM_PREPROCESSED_FILE})
+
           set(VM_SOURCES_WKS_ARM64_PREPROCESSED_ASM ${VM_SOURCES_WKS_ARM64_PREPROCESSED_ASM} ${ASM_PREPROCESSED_FILE})
+
       endforeach()
 
       set(VM_SOURCES_WKS_ARCH_ASM ${VM_SOURCES_WKS_ARM64_PREPROCESSED_ASM})
+
+  elseif(CLR_CMAKE_PLATFORM_ARCH_ARM)
+
+      get_include_directories_asm(ASM_INCLUDE_DIRECTORIES)
+
+      foreach(ASM_FILE ${VM_SOURCES_WKS_ARCH_ASM})
+
+          # Inserts a custom command in CMake build to preprocess each asm source file
+          get_filename_component(name ${ASM_FILE} NAME_WE)
+          file(TO_CMAKE_PATH "$ENV{__IntermediatesDir}/${name}.asm" ASM_PREPROCESSED_FILE)
+          preprocess_def_file(${ASM_FILE} ${ASM_PREPROCESSED_FILE})
+
+          # On Arm32, compile the preprocessed binary to .obj
+          # We do not pass any defines since we have already done pre-processing above
+          set (ASM_CMDLINE "-o ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj ${ASM_PREPROCESSED_FILE}")
+
+          # Generate the batch file that will invoke the assembler
+          file(TO_CMAKE_PATH "${CMAKE_CURRENT_BINARY_DIR}/runasm_${name}_${CMAKE_BUILD_TYPE}.cmd" ASM_SCRIPT_FILE)
+
+          # ARM32TODO: Set asm options global so that all invocations can use it.
+          file(GENERATE OUTPUT "${ASM_SCRIPT_FILE}"
+                    CONTENT "\"${CMAKE_ASM_MASM_COMPILER}\" -g ${ASM_INCLUDE_DIRECTORIES} ${ASM_DEFINITIONS} ${ASM_CMDLINE}")
+
+          message("Generated  - ${ASM_SCRIPT_FILE}")
+
+          # Need to compile asm file using custom command as include directories are not provided to asm compiler
+          add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj
+                             COMMAND ${ASM_SCRIPT_FILE}
```

Comment:
Same here can we use the command directly instead of creating temporary .cmd file